docs/providers.md

```diff
@@ -0,0 +1,114 @@
+* [Microsoft Azure][provider-azure]
+  * [API Reference][azure-reference]
+* [Rook][provider-rook]
+  * [API Reference][rook-reference]
+* [Alibaba Cloud][provider-alibaba]
+  * [API Reference][alibaba-reference]
+
+## Installation
+
+Since all managed resources are all cluster-scoped, provider packages are
+bundled as `ClusterPackage`s. In order to request installation of a provider,
+the following `ClusterPackageInstall` needs to be created:
+
+```yaml
+apiVersion: packages.crossplane.io/v1alpha1
+kind: ClusterPackageInstall
+metadata:
+  name: provider-gcp
+  namespace: crossplane-system
+spec:
+  package: "crossplane/provider-gcp:master"
+```
+
+The field `spec.package` is where you refer to the container image of the
+provider. Crossplane Package Manager will unpack that container, register
+CRDs and set up necessary RBAC rules and then start the controllers.
+
+There are a few ways that are more convenient than creating a YAML to install
+the providers.
+
+* As part of Crossplane Helm chart by adding the following
+  statement to your `helm install` command: `--set clusterPackages.gcp.deploy=true`
+  It will install the default version hard-coded in that release of Crossplane
+  Helm chart but if you'd like to specif an exact version , you can use:
+  `--set clusterPackages.gcp.version=master`.
+* Using Crossplane kubectl plugin:
+  `kubectl crossplane package install --cluster -n crossplane-system 'crossplane/provider-gcp:master' provider-gcp`
+
+You can delete them by deleting the installed `ClusterPackageInstall` resource.
+
+## Configuration
+
+In order to authenticate with the provider API, the provider controllers
+need to have access to credentials. It could be an AWS IAM User, GCP Service
+Account or Azure Service Principal. Every managed resource like `RDSInstance`
+or `CloudSQLInstance` has a field called `spec.providerRef` to refer to what
+credentials to use for operations regarding that managed resource. That
+reference points to a specific type called `Provider` that contains information
+about the authentication method that should be used. A `Provider` resource for
+Azure looks like the following:
+
+```yaml
+apiVersion: azure.crossplane.io/v1alpha3
+kind: Provider
+metadata:
+  name: prod-acc
+spec:
+  credentialsSecretRef:
+    namespace: crossplane-system
+    name: azure-prod-creds
+    key: credentials
+```
+
+You can see that there is a reference to a key in a specific `Secret`. The value
+of that key should contain the credentials that the controller will use. The
+documentation of each provider should give you an idea of how that credentials
+blob should look like.
+
+Keep in mind that it's not a requirement that the authentication should be done
+through a credentials blob in a `Secret`. For example, in AWS, `Provider` has
+a field called `useServiceAccount` to imply that the provider controller
+should use workload identity feature of AWS instead of a `Secret` that contains
+IAM credentials.
```

Comment:
Link? https://github.com/crossplane/provider-aws/blob/b3608d666b95b65387dc3802117d01a15959e61e/AUTHENTICATION.md

Do we have equivalent workload identity options for other providers?

Reply:
I believe we only support this for AWS today. It's a somewhat underdeveloped experience that we put into place quickly because our several of users were asking for it. I'd suggest including it as a footnote - i.e. one sentence with a link. Something like:

> The AWS provider includes experimental support for authenticating using a Kubernete service account. See [the documentation for details](https://github.com/crossplane/provider-aws/blob/b3608d666b95b65387dc3802117d01a15959e61e/AUTHENTICATION.md).

Reply:
This also currently requires some additional work due to not being able to run containers as `root`, which is required to access the token that AWS EKS webhook injects. It can still be done but requires doing your own build of `provider-aws`. The current thinking is to wait until this is fixed with projected service account volume in upstream k8s. More info here: https://github.com/crossplane/provider-aws/issues/211

Reply:
If it's not working as is, I'd prefer removing it. What do you think?

Reply:
@muvaf yeah I think that sounds good for now :+1: